src/v/cloud_storage/remote_segment_index.h

```diff
@@ -108,8 +110,22 @@ class offset_index {
     _find_under(deltafor_decoder<int64_t> decoder, int64_t offset);
 
     /// Return element by index.
+    template<class DecoderT>
     static std::optional<int64_t>
-    _fetch_ix(deltafor_decoder<int64_t> decoder, size_t target_ix);
+    _fetch_ix(DecoderT decoder, size_t target_ix) {
+        size_t ix = 0;
+        std::array<int64_t, buffer_depth> buffer{};
+        while (decoder.read(buffer)) {
+            for (auto o : buffer) {
+                if (ix == target_ix) {
+                    return o;
+                }
+                ix++;
+            }
+            buffer = {};
+        }
+        return std::nullopt;
+    }
```

Comment:
just curious, why did you move the implementation to `.h` file?

Reply:
Because it's a template. It is possible to move implementation to .cc file
but it will require explicit template instantiation.